bin/git-undo

```diff
@@ -12,26 +33,40 @@ EOL
     read -r res
     case "${res}" in
       "Y" | "y")
-        test $2 -gt 1 > /dev/null 2>&1 && back="~$2"
-        git reset --hard HEAD$back
-        exit $?
+        parm1=hard
+        parm2=${2:-1}
         ;;
       * )
         exit 0
         ;;
     esac
     ;;
   --hard)
-    test $2 -gt 1 > /dev/null 2>&1 && back="~$2"
-    git reset --hard HEAD$back
+    parm1=hard
+    parm2=${2:-1}
     ;;
   -s|--soft)
-    test $2 -gt 1 > /dev/null 2>&1 && back="~$2"
-    git reset --soft HEAD$back
+    parm1=soft
+    parm2=${2:-1}
+    parm3=reset
+    ;;
+  "")
+    parm1=soft
+    parm2=1
+    ;;
+  *[!0-9]*)
+    echo "Invalid parameter: $1"
+    exit 1
     ;;
   *)
-    test $1 -gt 1 > /dev/null 2>&1 && back="~$1"
-    git reset --soft HEAD$back
-    git reset
+    parm1=soft
+    parm2="$1"
     ;;
 esac
+
+if [[ ! $parm2 =~ [1-9]*([0-9]) ]]; then
```

Comment:
I have changed my mind. Since `0` is not a valid count, maybe the pattern should be `[1-9][0-9]*`?

Reply:
That's not really changing your mind, that's what you brought up initially, which is why I changed the first one to be 1-9 instead of 0-9. What I have does exactly what you're suggesting, using [[ regex syntax.

Reply:
@vr8hub
Err... This regex pattern is better: `if [[ ! $parm2 =~ ^[1-9][0-9]*$ ]]`.
It can detect all three invalid cases:`0`, `1s`, `01`.

Reply:
Done.